packages/@react-aria/dialog/src/useDialog.ts

```diff
@@ -62,7 +67,7 @@ export function useDialog(props: AriaDialogProps, ref: RefObject<HTMLElement>):
     dialogProps: {
       ...filterDOMProps(props, {labelable: true}),
       role,
-      tabIndex: -1,
+      tabIndex: (autoFocused ? -1 : 0),
```

Comment:
We can get rid of the autoFocused logic if https://github.com/adobe/react-spectrum/pull/2763 goes in right?

Reply:
I don't think we can get rid of this logic. #2763 handles the case when there are no focusable elements within the scope. The `autoFocused` logic here handles the case when there may be _tabbable_ elements within the scope, but we want the Dialog itself, a _focusable_ element, to receive focus.

Reply:
Hm, I see, it's necessary since `useAutoFocus`'s useLayoutEffect triggers before the `useDialog`'s useLayoutEffect, moving focus to the button. Remind me, what prompted the change to make `useAutoFocus` a `useLayoutEffect` instead of a `useEffect`?